include/btree/chunk_ops.h

```diff
@@ -0,0 +1,129 @@
+/*-------------------------------------------------------------------------
+ *
+ * chunk_ops.h
+ *		OrioleDB abstract page chunk access API.
+ *
+ * Copyright (c) 2024, Oriole DB Inc.
+ *
+ * IDENTIFICATION
+ *	  contrib/orioledb/include/btree/chunk_ops.h
+ *
+ *-------------------------------------------------------------------------
+ */
+#ifndef __BTREE_CHUNK_OPS_H__
+#define __BTREE_CHUNK_OPS_H__
+
+#include "btree/btree.h"
+#include "btree/page_contents.h"
+
+typedef enum BTreeChunkOperationType
+{
+	BTreeChunkOperationInsert,
+	BTreeChunkOperationUpdate,
+	BTreeChunkOperationDelete
+} BTreeChunkOperationType;
+
+typedef enum BTreeChunkType
+{
+	BTreeChunkHiKeys,
+	BTreeChunkNonLeaf,
+	BTreeChunkLeaf
+} BTreeChunkType;
+
+typedef struct BTreeChunkBuilderData
```

Comment:
I think we need a pre-allocated array of items (like BTreePageChunk.items) here. First, fill them with offsets of tuples from the beginning of the first tuple. On build finish, advance them with the size of the items array itself.

Reply:
Added new fields `chunkItems` and `chunkItemsCount`.